From cce5d816779cf8a941a6e7662c5f78c84b8519ee Mon Sep 17 00:00:00 2001
From: Steven Barrett <steven@liquorix.net>
Date: Wed, 22 Jan 2020 15:31:59 -0600
Subject: [PATCH 15/15] exfat: Depend on EXFAT_FS_SAMSUNG=n from mainline exfat
 filesystem

Adding a "depends on" from EXFAT_FS_SAMSUNG for !EXFAT_FS ended up some
how allowing both EXFAT_FS + EXFAT_FS_SAMSUNG.  Modifying our dependency
test to look for EXFAT_FS=n seemed to do better, but "make oldconfig"
didn't ask us if we want EXFAT_FS_SAMSUNG enabled.

Lets move the test to fs/exfat/Kconfig so that we're asked about the new
exfat filesystem on our next run of "make oldconfig".
---
 drivers/staging/exfat/Kconfig | 2 +-
 fs/exfat/Kconfig              | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/exfat/Kconfig b/drivers/staging/exfat/Kconfig
index 41bffe4a53c8..9e3d663b5f13 100644
--- a/drivers/staging/exfat/Kconfig
+++ b/drivers/staging/exfat/Kconfig
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 config EXFAT_FS_SAMSUNG
 	tristate "exFAT fs support"
-	depends on BLOCK && !EXFAT_FS
+	depends on BLOCK
 	select NLS
 	help
 	  This adds support for the exFAT file system.
diff --git a/fs/exfat/Kconfig b/fs/exfat/Kconfig
index 2d3636dc5b8c..6da9cc67621f 100644
--- a/fs/exfat/Kconfig
+++ b/fs/exfat/Kconfig
@@ -3,6 +3,7 @@
 config EXFAT_FS
 	tristate "exFAT filesystem support"
 	select NLS
+	depends on EXFAT_FS_SAMSUNG=n
 	help
 	  This allows you to mount devices formatted with the exFAT file system.
 	  exFAT is typically used on SD-Cards or USB sticks.
-- 
2.25.0.2.g232378479e.dirty

